tests/functional/ingestion/IngestionReader.js

```diff
@@ -0,0 +1,247 @@
+        this.ingestionReader = new IngestionReader({
+            zkClient,
+            kafkaConfig: testConfig.kafka,
+            bucketdConfig: testConfig.extensions.ingestion.sources[0],
+            qpConfig: testConfig.queuePopulator,
+            logger: dummyLogger,
+            extensions: [extIngestionQP],
+            s3Config: testConfig.s3,
+            bucket: testConfig.extensions.ingestion.sources[0].bucket,
+        });
+        this.ingestionReader.setup(() => {
+            zkClient.setData(this.ingestionReader.pathToLogOffset,
+                Buffer.from('2'), -1, err => {
+                    assert.ifError(err);
+                    return done();
+                });
+        });
+    });
+
+    afterEach(done => {
+        httpServer.close();
+        done();
+    });
+
+    after(done => {
+        this.db.collection('PENSIEVE').drop(err => {
+            assert.ifError(err);
+            this.client.close();
+            done();
+        });
+    });
+
+    it('_processReadRecords should retrieve logRes stream', done => {
+        assert.strictEqual(batchState.logRes, null);
+        return this.ingestionReader._processReadRecords({}, batchState, err => {
+            assert.ifError(err);
+            assert.deepStrictEqual(batchState.logRes.info,
+                { start: 1, cseq: 7, prune: 1 });
+            const receivedLogs = [];
+            batchState.logRes.log.on('data', data => {
+                receivedLogs.push(data);
+            });
+            batchState.logRes.log.on('end', () => {
+                assert.strictEqual(receivedLogs.length, 7);
+                return done();
+            });
+        });
+    });
+
+    // Assertion on parsedlogs here is done in the extIngestionQP mock
+    it('_processPrepareEntries should send entries in the correct format and ' +
+    'update `nbLogEntriesRead` + `nbLogRecordsRead`', done => {
+        async.waterfall([
+            next =>
+                this.ingestionReader._processReadRecords({}, batchState, next),
+            next =>
+            this.ingestionReader._processPrepareEntries(batchState, next),
+        ], () => {
+            assert.deepStrictEqual(batchState.logStats, {
+                nbLogRecordsRead: 7, nbLogEntriesRead: 7,
+            });
+            return done();
+        });
+    });
+
+    it('should successfully run setup()', done => {
+        this.ingestionReader.setup(err => {
+            assert.ifError(err);
+            return done();
+        });
+    });
+
+    it('should get logOffset', done => {
+        const logOffset = this.ingestionReader.getLogOffset();
+        // value initialized when creating MockZkClient
+        assert.equal(logOffset, mockLogOffset);
+        done();
+    });
+
+    it('should succesfully ingest new bucket with existing objects', done => {
+        async.waterfall([
+            next => zkClient.setData(this.ingestionReader.pathToLogOffset,
+                Buffer.from('1'), -1, err => {
```

Comment:
As discussed, you can add a comment here that reading from logOffset 1 would mean ingesting existing objects.